### PR TITLE
refactor: link selector pass functions

### PR DESCRIPTION
**Pull request by ragi96**
to decouple the linkSelector further from the appBridge, we will pass the functions and not the appBridge itself

CU-8693rnyhk

### CODE REVIEW

File: .changeset/loud-badgers-fly.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+```
+
+with:
+
+```jsx
+<LinkSelector
+    url={url}
+    onUrlChange={onUrlChange}
+    buttonSize={buttonSize ?? ButtonSize.Medium}
+    getAllDocuments={appBridge.getAllDocuments}
```
Comment (fulopdaniel) [inline .changeset/loud-badgers-fly.md:25]: `bind` missing from the changeset?

Comment (ragi96) [reply .changeset/loud-badgers-fly.md]: improved the syntax so we don't need the `bind` and adjusted it here

File: packages/guideline-blocks-settings/src/components/Link/LinkSelector/DocumentLinks.tsx [hunk 2]
```diff
@@ -52,16 +59,16 @@ export const DocumentLinks = ({ appBridge, selectedUrl, onSelectUrl }: DocumentL
             return itemsToExpand;
         }
         for (const document of documentArray) {
-            const pages = await appBridge.getDocumentPagesByDocumentId(document.id);
-            appBridge.getAllDocuments();
+            const pages = await getDocumentPagesByDocumentId(document.id);
+            getAllDocuments();
```
Comment (fulopdaniel) [inline packages/guideline-blocks-settings/src/components/Link/LinkSelector/DocumentLinks.tsx:63]: I think this call was here by mistake. We can remove it

Comment (ragi96) [reply packages/guideline-blocks-settings/src/components/Link/LinkSelector/DocumentLinks.tsx:63]: Seems like it, and stupid me didn't saw it, thanks

**Review by fulopdaniel — APPROVED** [review]

**Pull request merged**